diff --git a/node_modules/bpmn-js-token-simulation/lib/features/context-pads/handler/PauseHandler.js b/node_modules/bpmn-js-token-simulation/lib/features/context-pads/handler/PauseHandler.js
index 2e7d232..057e8a4 100644
--- a/node_modules/bpmn-js-token-simulation/lib/features/context-pads/handler/PauseHandler.js
+++ b/node_modules/bpmn-js-token-simulation/lib/features/context-pads/handler/PauseHandler.js
@@ -10,6 +10,18 @@ export default function PauseHandler(simulator) {
 
 PauseHandler.prototype.createContextPads = function(element) {
 
+  if ( element.businessObject.triggeredByEvent || 
+       element.type == 'bpmn:ReceiveTask' || 
+       ( element.type == 'bpmn:Task' && 
+         ( element.businessObject.type == 'Resource'  
+           || element.businessObject.type == 'Request' 
+           || element.businessObject.type == 'Release' 
+         )
+       )
+     ) {
+     return [] ;
+  }
+
   return [
     this.createPauseContextPad(element)
   ];
diff --git a/node_modules/bpmn-js-token-simulation/lib/features/log/Log.js b/node_modules/bpmn-js-token-simulation/lib/features/log/Log.js
index ac0322e..80f941b 100644
--- a/node_modules/bpmn-js-token-simulation/lib/features/log/Log.js
+++ b/node_modules/bpmn-js-token-simulation/lib/features/log/Log.js
@@ -199,11 +199,35 @@ export default function Log(
         scope
       });
     } else if (is(element, 'bpmn:Task')) {
-      this.log({
-        text: elementName || 'Task',
-        icon: 'bpmn-icon-task',
-        scope
-      });
+      const bo = getBusinessObject(element);
+      if ( bo.type == 'Request' ) {
+        this.log({
+          text: elementName || 'RequestTask',
+          icon: 'bpmn-icon-request',
+          scope
+        });
+      }
+      else if ( bo.type == 'Release' ) {
+        this.log({
+          text: elementName || 'ReleaseTask',
+          icon: 'bpmn-icon-release',
+          scope
+        });
+      }
+      else if ( bo.type == 'Resource' ) {
+        this.log({
+          text: elementName || 'ResourceTask',
+          icon: 'bpmn-icon-resource',
+          scope
+        });
+      }
+      else {
+        this.log({
+          text: elementName || 'Task',
+          icon: 'bpmn-icon-task',
+          scope
+        });
+      }
     } else if (is(element, 'bpmn:UserTask')) {
       this.log({
         text: elementName || 'User Task',
diff --git a/node_modules/bpmn-js-token-simulation/lib/simulator/Simulator.js b/node_modules/bpmn-js-token-simulation/lib/simulator/Simulator.js
index ff50389..203ad69 100644
--- a/node_modules/bpmn-js-token-simulation/lib/simulator/Simulator.js
+++ b/node_modules/bpmn-js-token-simulation/lib/simulator/Simulator.js
@@ -56,6 +56,13 @@ export default function Simulator(injector, eventBus) {
   }
 
   function getBehavior(element) {
+    if ( element.type == 'bpmn:Task' && 
+         ( element.businessObject.type == 'Resource'  
+           || element.businessObject.type == 'Request' 
+           || element.businessObject.type == 'Release' )
+       ) {
+       return behaviors['bpmn:ReceiveTask'];
+    }
     return behaviors[element.type] || noopBehavior;
   }
 
