diff --git a/node_modules/bpmn-js-properties-panel/lib/PropertiesPanel.js b/node_modules/bpmn-js-properties-panel/lib/PropertiesPanel.js
index b011956..6802ff2 100644
--- a/node_modules/bpmn-js-properties-panel/lib/PropertiesPanel.js
+++ b/node_modules/bpmn-js-properties-panel/lib/PropertiesPanel.js
@@ -329,6 +329,9 @@ PropertiesPanel.prototype._init = function(config) {
       if (e.elements.indexOf(element) !== -1) {
         self.update(element);
       }
+      else if (e.elements[0].$type == 'bpmn:Definitions') {
+        self.update(e.elements[0]);
+      }
     }
   });
